fs.js

```diff
@@ -0,0 +1,278 @@
+			return new Writable({write: (chunk, encoding, callback) => callback()});
+		}
+		return fs.createWriteStream(this.path, options);
+	}
+	createAppendStream(options = {}) {
+		if (Config.nofswriting) {
+			const Writable = require('stream').Writable;
+			return new Writable({write: (chunk, encoding, callback) => callback()});
+		}
+		options.mode = options.mode || 'a';
+		return fs.createWriteStream(this.path, options);
+	}
+	unlinkIfExists() {
+		if (Config.nofswriting) return Promise.resolve();
+		return new Promise((resolve, reject) => {
+			fs.unlink(this.path, err => {
+				if (err && err.code === 'ENOENT') return resolve();
+				err ? reject(err) : resolve();
+			});
+		});
+	}
+	unlinkIfExistsSync() {
+		if (Config.nofswriting) return;
+		try {
+			fs.unlinkSync(this.path);
+		} catch (err) {
+			if (err.code !== 'ENOENT') throw err;
+		}
+	}
+	/**
+	 * @param {string | number} mode
+	 */
+	mkdir(mode = 0o755) {
+		if (Config.nofswriting) return Promise.resolve();
+		return new Promise((resolve, reject) => {
+			// @ts-ignore
+			fs.mkdir(this.path, mode, err => {
+				err ? reject(err) : resolve();
+			});
+		});
+	}
+	/**
+	 * @param {string | number} mode
+	 */
+	mkdirSync(mode = 0o755) {
+		if (Config.nofswriting) return;
+		// @ts-ignore
+		return fs.mkdirSync(this.path, mode);
+	}
+	/**
+	 * @param {string | number} mode
+	 */
+	mkdirIfNonexistent(mode = 0o755) {
+		if (Config.nofswriting) return Promise.resolve();
+		return new Promise((resolve, reject) => {
+			// @ts-ignore
+			fs.mkdir(this.path, mode, err => {
+				if (err && err.code === 'EEXIST') return resolve();
+				err ? reject(err) : resolve();
+			});
+		});
+	}
+	/**
+	 * @param {string | number} mode
+	 */
+	mkdirIfNonexistentSync(mode = 0o755) {
+		if (Config.nofswriting) return;
+		try {
+			// @ts-ignore
+			fs.mkdirSync(this.path, mode);
+		} catch (err) {
+			if (err.code !== 'EEXIST') throw err;
+		}
+	}
+	/**
+	 * Creates the directory (and any parent directories if necessary).
+	 * Does not throw if the directory already exists.
+	 * @param {string | number} mode
+	 */
+	async mkdirp(mode = 0o755) {
+		try {
+			await this.mkdirIfNonexistent(mode);
+		} catch (err) {
+			if (err.code !== 'ENOENT') throw err;
+			await this.parentDir().mkdirp(mode);
+			await this.mkdirIfNonexistent(mode);
+		}
+	}
+	/**
+	 * Creates the directory (and any parent directories if necessary).
+	 * Does not throw if the directory already exists. Synchronous.
+	 * @param {string | number} mode
+	 */
+	mkdirpSync(mode = 0o755) {
+		try {
+			this.mkdirIfNonexistentSync(mode);
+		} catch (err) {
+			if (err.code !== 'ENOENT') throw err;
+			this.parentDir().mkdirpSync(mode);
+			this.mkdirIfNonexistentSync(mode);
+		}
+	}
+	/**
+	 * Calls the callback if the file is modified.
+	 * @param {function (): void} callback
+	 */
+	onModify(callback) {
+		fs.watchFile(this.path, (curr, prev) => {
+			if (curr.mtime > prev.mtime) return callback();
+		});
```

Comment:
Don't we override config during this?

Anyway yeah, `watchconfig` should probably just be `false`